src/lib/operators/aggregate_sort.cpp

```diff
@@ -368,14 +427,41 @@ std::shared_ptr<const Table> AggregateSort::_on_execute() {
    * However, we did not benchmark it, so we cannot prove it.
    */
 
-  // Sort input table consecutively by the group by columns (stable sort)
-  auto sorted_table = input_table;
-  for (const auto& column_id : _groupby_column_ids) {
-    const auto sorted_wrapper = std::make_shared<TableWrapper>(sorted_table);
-    sorted_wrapper->execute();
-    Sort sort = Sort(sorted_wrapper, column_id);
-    sort.execute();
-    sorted_table = sort.get_output();
+  auto table_value_clustered_by = input_table->value_clustered_by();
+
+  std::shared_ptr<const Table> sorted_table = input_table;
+  if (!_groupby_column_ids.empty()) {
+    /**
+     * Check if the value clustering matches the columns we are grouping by
+     * (these are the columns we would need to sort by to achieve value clustering).
+     * If the input is already value clustered according to our needs, this means that values that would end up
+     * in the same cluster are already in the same chunk.
+     * Thus, we can execute all remaining sorts on a per-chunks basis.
+     * If the value clustering doesn't match the columns we group by, we need to re-sort the whole table
+     * to achieve the value clustering we need.
+     */
+    if (table_value_clustered_by && !table_value_clustered_by->empty() &&
+        std::find(_groupby_column_ids.begin(), _groupby_column_ids.end(), *(table_value_clustered_by->begin())) !=
+            _groupby_column_ids.end()) {
+      // Sort input table chunk_wise consecutively by the group by columns (stable sort)
+      sorted_table = _sort_table_chunk_wise(input_table, table_value_clustered_by);
+    } else {
+      // sort input table in whole consecutively by the group by columns
+      for (const auto& column_id : _groupby_column_ids) {
+        auto table_wrapper = std::make_shared<TableWrapper>(sorted_table);
+        table_wrapper->execute();
+        auto sort = std::make_shared<Sort>(table_wrapper, column_id);
+        sort->execute();
+        sorted_table = sort->get_output();
+      }
+      auto last_chunk = sorted_table->last_chunk();
+      if (last_chunk->is_mutable()) {
+        last_chunk->finalize();
+      }
+      // If the last column is sorted here, it needs to be set accordingly.
+      // This overrides order_by on purpose (only the last order prevails).
+      last_chunk->set_ordered_by(std::make_pair(_groupby_column_ids.back(), OrderByMode::Ascending));
```

Comment:
You are right, the current state is very misleading. I changed `set_ordered_by`'s logic so that it indeed only sets the ordered information. Nothing is added anymore. If a chunk is ordered by multiple columns, the `void set_ordered_by(std::vector)` signature should be used. Is that ok for you? I also extended the corresponding test in `chunk_test` to ensure this behavior.

Concerning a possible error: I do not think that it is an error in `aggregate_sort.cpp`. Sorting information could only be set when table is sorted before. The sorting leads to a table with one single chunk. The sorting information is always set to the `_groupby_column_ids` so nothing is added here, only updated

Reply:
Perfect.